### PR TITLE
ServiceNow Plugin : Make PAGE_SIZE configurable

### PR DESCRIPTION
**Pull request by sgarg-CS**
[PLUGIN-1689](https://cdap.atlassian.net/browse/PLUGIN-1689) : Make Page Size configurable in Service Now Batch Source and Multi-Source Plugins

[PLUGIN-1689]: https://cdap.atlassian.net/browse/PLUGIN-1689?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java [hunk 1]
```diff
@@ -208,6 +221,17 @@ private void validateDateRange(FailureCollector collector) {
     }
   }
 
+  private void validatePageSize(FailureCollector collector) {
+    if (containsMacro(ServiceNowConstants.PROPERTY_PAGE_SIZE)) {
+      return;
+    }
+
+    if (getPageSize() < 0 || getPageSize() > 10000) {
```
Comment (saimukkamala) [inline src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java]: should be `if (getPageSize() <= 0 || getPageSize() > 10000) {`? `int pages = (tableInfo.getRecordCount() / pageSize);` - this line would divide by 0.

Comment (sgarg-CS) [reply src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java]: Good catch! 
Refactored it. 84fa2da26d0a11988d339f44dd6c112e87cff819

File: src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java [hunk 2]
```diff
@@ -208,6 +221,18 @@ private void validateDateRange(FailureCollector collector) {
...
   }
 
+  private void validatePageSize(FailureCollector collector) {
+    if (containsMacro(ServiceNowConstants.PROPERTY_PAGE_SIZE)) {
+      return;
+    }
+
+    if (getPageSize() <= 0 || getPageSize() > 10000) {
+      collector.addFailure("Invalid Page Size.", "Page Size must be greater than 0 and " +
+          "less than 10000.")
```
Comment (saimukkamala) [inline src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java]: less than or equal to 10000

**Review by saimukkamala — APPROVED** [review]

File: src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java (hunk 2 above)
Comment (sgarg-CS) [reply src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java]: 10000 is a valid Page Size.
Update the failure message. de8166ce721ff42bb0fa51b4d9936fa12f1088e3

**Pull request merged**